run-build.ps1

```diff
@@ -110,11 +110,11 @@ $env:PATH = "$env:DOTNET_INSTALL_DIR;$env:PATH"
 if ($NoBuild)
 {
     Write-Host "Not building due to --nobuild"
-    Write-Host "Command that would be run: 'dotnet msbuild build.proj /m /p:Architecture=$Architecture $ExtraParameters'"
+    Write-Host "Command that would be run: 'dotnet msbuild build.proj /m /p:OverrideArchitecture=$Architecture $ExtraParameters'"
 }
 else
 {
-    dotnet msbuild build.proj /p:Architecture=$Architecture /p:GeneratingPropsFile=true /t:WriteDynamicPropsToStaticPropsFiles
-    dotnet msbuild build.proj /m /v:diag /p:Architecture=$Architecture $ExtraParameters
+    dotnet msbuild build.proj /p:OverrideArchitecture=$Architecture /p:GeneratingPropsFile=true /t:WriteDynamicPropsToStaticPropsFiles
```

Comment:
Properties passed in via `/p:PropName=value` on the command line are immutable within the build, so @eerhardt's right that they always* win.

* There is an exception: if the project specifies that a property is one of its `TreatAsLocalProperties`, then it can be changed within the build. That's rare and not the default.